From 7596e7792cb59a129f06ff41c4da03baa4d64b96 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 27 Apr 2018 12:34:00 +1000
Subject: [PATCH] d3d11: D3D11_USAGE_DYNAMIC needs to have CPU access

Fix regression of b6f917b1023ea2b9591f6b72d34ff2afa34bd914

Reference  (details on D3D11_USAGE_DYNAMIC usage)
https://msdn.microsoft.com/en-us/library/windows/desktop/ff476259(v=vs.85).aspx


Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/d3d11/buffer.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/d3d11/buffer.c b/dlls/d3d11/buffer.c
index 09778e2..04d473b 100644
--- a/dlls/d3d11/buffer.c
+++ b/dlls/d3d11/buffer.c
@@ -453,6 +453,11 @@ static HRESULT d3d_buffer_init(struct d3d_buffer *buffer, struct d3d_device *dev
     wined3d_desc.usage = wined3d_usage_from_d3d11(0, buffer->desc.Usage);
     wined3d_desc.bind_flags = buffer->desc.BindFlags;
     wined3d_desc.access = wined3d_access_from_d3d11(buffer->desc.Usage, buffer->desc.CPUAccessFlags);
+    if(buffer->desc.Usage == D3D11_USAGE_DYNAMIC)
+    {
+        TRACE("Setting Dynamic CPU access\n");
+        wined3d_desc.access |= WINED3D_RESOURCE_ACCESS_CPU;
+    }
     wined3d_desc.misc_flags = buffer->desc.MiscFlags;
     wined3d_desc.structure_byte_stride = buffer->desc.StructureByteStride;
 
-- 
1.9.1

